### PR TITLE
code quality checker

### PR DESCRIPTION
**Pull request by superseacat**
the stats are being pushed to sonarcloud: https://sonarcloud.io/dashboard?branch=code-quality-checker&id=Aalto-LeTech_intellij-plugin , but as there is no code yet it shows empty. 

there is still a problem of travis not building all the time properly, but I hope merging to master (as master does not have yet a travis.yml) might solve it.

please review, check and comment

### CODE REVIEW

File: travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  - oraclejdk11
+
+addons:
+  sonarcloud:
+    organization: "aalto-letech-intellij-plugin"
+    token: $SONARCLOUD_TOKEN
+script:
+  - sonar-scanner
+branches:
+  - /.*/
```
Comment (OlliKiljunen) [inline travis.yml]: I guess this means build all braches. In <https://docs.travis-ci.com/user/customizing-the-build/#Skipping-the-Installation-Step> they use a bit longer form for that.
```
branches:
  only:
  - /.*/
```
I assume this is a proper shorthand as Travis build config is correctly constructed.

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+sonar.projectKey=Aalto-LeTech_intellij-plugin
+sonar.organization="aalto-letech-intellij-plugin"
+sonar.analysis.mode=publish
```
Comment (OlliKiljunen) [inline sonar-project.properties]: Should I nitpick of missing LF in the end of file. :D

File: travis.yml (hunk 1 above)
Comment (superseacat) [reply travis.yml]: travis build is triggered with a "workaround" (trigger build) feature of [travis](https://blog.travis-ci.com/2017-08-24-trigger-custom-build), so for now there has been no automated pick-ups from gh, which makes me puzzled

Comment (superseacat) [reply travis.yml]: FFS, it's **.travis.yml** (starting with a dot) but not **travis.yml**

Comment (superseacat) [reply travis.yml]: https://travis-ci.org/Aalto-LeTech/intellij-plugin/builds/634191182 like a charm

File: .travis.yml [hunk 3]
```diff
@@ -10,9 +10,5 @@ addons:
 script:
   - sonar-scanner
   
-branches:
-  only:
-  - /.*/
-
```
Comment (OlliKiljunen) [inline .travis.yml]: Oh yeah, no mention about branches means also that all branches are built. Good catch.

**Pull request merged**